binchihua/src/client/client.c

```diff
@@ -1,7 +1,32 @@
+#include <signal.h>
+#include <stdbool.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <unistd.h>
+
+// NOLINTNEXTLINE
+static bool done = false;
+
+static void signal_handler(int signal) {
+        printf("\n%s(): Received signal %d\n", __func__, signal);
+        done = true;
+}
 
 int main() {
-        fprintf(stdout, "Hello from client!\n");
-        return EXIT_SUCCESS;
-}
\ No newline at end of file
+        struct sigaction sa;
```

Comment:
I think it would be great to move this to a dedicated function, e.g. `register_shutdown_handler()`, and pass in only the handler function. This way we can reuse this also for the node and orchestrator.

Reply:
That's a good idea to have a signal handler that is common across our three apps.  Could we come back to this step later, as part of implementing shutdown?  For this task, I mainly wanted to provide an example of a signal handler.